sdk/servicebus/azure-servicebus/tests/test_sessions.py

```diff
@@ -908,12 +909,16 @@ def test_session_by_session_client_conn_str_receive_handler_peeklock_abandon(sel
                     message = Message("Handler message no. {}".format(i), session_id=session_id)
                     sender.send(message)
 
-            with sb_client.get_queue_session_receiver(servicebus_queue.name, session_id=session_id, prefetch=0) as receiver:
+            with sb_client.get_queue_session_receiver(servicebus_queue.name, session_id=session_id, prefetch=0, idle_timeout=1) as receiver:
                 message = receiver.next()
                 assert message.sequence_number == 1
                 message.abandon()
-                second_message = receiver.next()
-                assert second_message.sequence_number == 1
+                while True: # we can't be sure there won't be a service delay, so we may not get the message back _immediately_, even if in most cases it shows right back up.
+                    second_message = receiver.next()
+                    if not second_message:
+                        raise "Did not successfully re-receive abandoned message, sequence_number 1 was not observed."
```

Comment:
raise a string?

Reply:
The best part?  I validated the failure case locally but didn't look too hard at _what_ the final exception threw, so double-fail on my part.  Good catch; and great evidence for why I need to stop trying to write code while in architecture design meetings.